include/chopper/pack/hierarchical_binning.hpp

```diff
@@ -105,6 +105,15 @@ class hierarchical_binning
             throw std::runtime_error{"The filenames and kmer counts do not have the same length."};
     }
 
+    /*!\brief Computes and returns the number of additionally needed recursive levels when doing a merge.
+     * \param[in] num_user_bins The number of user bins in the merge
+     */
+    size_t get_needed_levels(size_t const num_user_bins)
+    {
+        double const levels = std::log(num_user_bins) / std::log(num_technical_bins);
+        return static_cast<size_t>(std::ceil(levels));
```

Comment:
I don't know if it would be worth it to store the result of `std::log(num_technical_bins)` as a class member and reuse it here.